oauth2_provider/tests/test_authorization.py

```diff
@@ -0,0 +1,107 @@
+"""
+Tests various flows through using the Authorization Endpoint.
+"""
+from urlparse import urlparse, parse_qs
+
+import ddt
+import provider.oauth2.models
+import provider.scope
+
+from oauth2_provider.tests.base import OAuth2TestCase, IDTokenTestCase
+
+
+@ddt.ddt
+class ImplicitFlowTestCase(IDTokenTestCase):
+    """
+    Test OIDC-related extensions to the OAuth2 Authorization flows
+    implemented by this package.
+    """
+
+    def _get_auth_payload(self, location):
+        """
+        DRY helper.  Extract auth parameters from the redirect location as
+        returned during the implicit flow.
+        """
+        self.assertTrue(location.startswith(self.auth_client.redirect_uri + '#'))
+        return {k: v[0] for k, v in parse_qs(urlparse(location).fragment).items()}
+
+    def _get_access_token(self, scope):
+        """
+        DRY helper.  Find the access token generated during our auth request.
+        """
+        return provider.oauth2.models.AccessToken.objects.get(
+            user=self.user, client=self.auth_client, scope=provider.scope.to_int(*scope.split())
+        )
+
+    @ddt.data(
+        ('openid', 'id_token token'),
+        ('openid', 'id_token'),
+        ('openid profile', 'id_token'),
+        ('openid profile email', 'id_token'),
+    )
+    @ddt.unpack
+    def test_oidc_implicit_flow(self, scope, response_type):
+        """
+        Ensure that valid permutations of scope and response_type result in
+        correct OIDC behavior under the implicit flow.
+        """
+        response = self.login_and_authorize(scope=scope, response_type=response_type, trusted=True)
+        self.assertEqual(302, response.status_code)
+        auth_payload = self._get_auth_payload(response['Location'])
+
+        # check state
+        self.assertEqual(auth_payload['state'], 'some_state')
+
+        # check scope
+        self.assertEqual(set(auth_payload['scope'].split()), set(scope.split()))
+
+        expected_token = self._get_access_token(scope)
```

Comment:
in the test we need to fetch the underlying access token even if we don't expect it to be included in the response.   whether or not it is included in the response is still checked below (L61-66)